OptimizelySDK.Tests/OptimizelyTest.cs

```diff
@@ -1109,5 +1120,350 @@ public void TestGetVariationBucketingIdAttribute()
+                It.IsAny<UserAttributes>(), featureVariableType)).Returns("non_integer_value");
+            OptimizelyMock.Setup(om => om.GetFeatureVariableValueForType(It.IsAny<string>(), variableKeyNull, It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), featureVariableType)).Returns<string>(null);
+
+            Assert.AreEqual(100, OptimizelyMock.Object.GetFeatureVariableInteger(featureKey, variableKeyInt, TestUserId, null));
+            Assert.AreEqual(null, OptimizelyMock.Object.GetFeatureVariableInteger(featureKey, variableKeyDouble, TestUserId, null));
+            Assert.AreEqual(null, OptimizelyMock.Object.GetFeatureVariableInteger(featureKey, variableNonInt, TestUserId, null));
+            Assert.AreEqual(null, OptimizelyMock.Object.GetFeatureVariableInteger(featureKey, variableKeyNull, TestUserId, null));
+        }
+
+        [Test]
+        public void TestGetFeatureVariableStringReturnTypecastedValue()
+        {
+            var featureKey = "featureKey";
+            var variableKeyString = "varString1";
+            var variableKeyIntString = "varString2";
+            var variableKeyNull = "varNull";
+            var featureVariableType = FeatureVariable.VariableType.STRING;
+
+            OptimizelyMock.Setup(om => om.GetFeatureVariableValueForType(It.IsAny<string>(), variableKeyString, It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), featureVariableType)).Returns("Test String");
+            OptimizelyMock.Setup(om => om.GetFeatureVariableValueForType(It.IsAny<string>(), variableKeyIntString, It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), featureVariableType)).Returns("123");
+            OptimizelyMock.Setup(om => om.GetFeatureVariableValueForType(It.IsAny<string>(), variableKeyNull, It.IsAny<string>(),
+                It.IsAny<UserAttributes>(), featureVariableType)).Returns<string>(null);
+
+            Assert.AreEqual("Test String", OptimizelyMock.Object.GetFeatureVariableString(featureKey, variableKeyString, TestUserId, null));
+            Assert.AreEqual("123", OptimizelyMock.Object.GetFeatureVariableString(featureKey, variableKeyIntString, TestUserId, null));
+            Assert.AreEqual(null, OptimizelyMock.Object.GetFeatureVariableString(featureKey, variableKeyNull, TestUserId, null));
+        }
+
+        #endregion // Test GetFeatureVariable<Type> TypeCasting
+
+        #region Test GetFeatureVariableValueForType method
+
+        // Should return null and log error message when arguments are null or empty.
+        [Test]
+        public void TestGetFeatureVariableValueForTypeGivenNullOrEmptyArguments()
+        {
+            var featureKey = "featureKey";
+            var variableKey = "variableKey";
+            var variableType = FeatureVariable.VariableType.BOOLEAN;
+
+            Assert.IsNull(Optimizely.GetFeatureVariableValueForType(null, variableKey, TestUserId, null, variableType));
```

Comment:
Please a dd a comment before each line. 
For example: //Passing user id as null
// Passing user id as empty string

Reply:
Done